### ISSUE TITLE
TypeError when using momepy.remove_false_nodes with shapely.geometry.linestring.LineString

### ISSUE THREAD

**Issue by anisotropi4** [pysal/momepy#472]
### Describe the problem

The `momepy.remove_false_nodes` function call errors with a `TypeError` when called with a `GeoSeries` `LineString` created with `GeoPandas 0.12.2` when using `Shapely 2.0`.

```python
momepy.remove_false_nodes(line)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/dev/momepy/venv/lib/python3.10/site-packages/momepy/preprocessing.py", line 224, in remove_false_nodes
    coords = pygeos.get_coordinates(geom)
  File "/home/dev/momepy/TIPLOC/venv/lib/python3.10/site-packages/pygeos/coordinates.py", line 134, in get_coordinates
    return lib.get_coordinates(
TypeError: One of the arguments is of incorrect type. Please provide only Geometry objects.
```

### Steps to reproduce

Executing the following raises a `TypeError`:
```python   
import os
os.environ['USE_PYGEOS'] = '0'
import geopandas
import momepy
from shapely import LineString
line = geopandas.GeoSeries(LineString([[0, 1], [2, 3]]))
momepy.remove_false_nodes(line)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/dev/momepy/venv/lib/python3.10/site-packages/momepy/preprocessing.py", line 224, in remove_false_nodes
    coords = pygeos.get_coordinates(geom)
  File "/home/dev/momepy/TIPLOC/venv/lib/python3.10/site-packages/pygeos/coordinates.py", line 134, in get_coordinates
    return lib.get_coordinates(
TypeError: One of the arguments is of incorrect type. Please provide only Geometry objects.
```

Whereas the following raising no error:
```python   
import os
os.environ['USE_PYGEOS'] = '1'
import geopandas
import momepy
from shapely import LineString
line = geopandas.GeoSeries(LineString([[0, 1], [2, 3]]))
momepy.remove_false_nodes(line)
```

### Versions of your packages

```
momepy=0.5.4
geopandas=0.12.2
libpysal=4.7.0
networkx=3.0
```

### Your operating system

Linux 5.15.0-60-generic #66-Ubuntu x86_64 GNU/Linux

### Additional context

A candidate resolution would be to replace references to `pygeos` in the `momepy/preprocessing.py` file with `shapely` and `tree.query_bulk` call with `tree.query`. Initial testing suggests this would work.

For backward compatibility a `try` `except` block may be needed to handle associated compatibility errors.

My plan would be to submit this as a fork.

**Comment by anisotropi4** [comment]
This may be related to [Refactor code for shapely 2.0](https://github.com/pysal/momepy/issues/443) raised by @martinfleis

**Comment by anisotropi4** [comment]
Based on the discussion in pull request #473, this is a Shapely 2.0 compatibility issue. Based on guidance [here](https://github.com/pysal/momepy/pull/473#issuecomment-1465730704) and noting the GeoPandas discussion [here](https://github.com/geopandas/geopandas/issues/2691#issuecomment-1375890344) I will close this bug as it is a known issue that will be addressed in a future.

**Comment by mcooper** [comment]
TLDR:

```
import os
os.environ['USE_PYGEOS'] = '1'
```
